en/05_Contributing/05_Coding_Conventions/01_PHP_Coding_Conventions.md

```diff
@@ -136,6 +136,133 @@ class MyClass extends Class
+- Use `UpperCamelCase` for class names, interfaces, traits and enums
+- Use `snake_case` for private static configuration properties.
+- Use `SCREAMING_SNAKE_CASE` for constants including class contants, and environment variables
+- Note that there will be some inconsistencies with legacy code where static methods have been a combination of `MyClass::camelCase()` and `MyClass::snake_case()`
+
+### Accessors
+
+- For accessors omit the leading 'get' with getters and prefix with 'set' for setters. e.g. `function myValue():sting { return $this->value; }` and `function setMyValue(string $value): void { $this->value = $value; }`.
+- Note that there will be some inconsistencies with legacy code as get accessors have been a combination of "with get", "without get", and "without get with the first letter uppercased"
+
+### File suffixes and prefixes
+
+- This is for new classes and interfaces that are created. Legacy Silvesrtripe code is inconsistent whether or not it follows these conventions.
+- Suffix the classnames with the type of class that is subclassed. The suffixed is sometimes a shortened version of the parent class because it reads better while retaining easy comprehension. Here are some common examples with the parents classes in brackets
+  - `Exception` (Exception)
+  - `Controller` (Controller)
+  - `Extension` (Extension, DataExtension)
+  - `Admin` (ModelAdmin)
+  - `Handler` (RequestHandler)
+  - `Filter` (SearchFilter)
+  - `Form` (Form)
+  - `Field` (FormField)
+  - `Job` (QueuedJob)
+  - `Task` (BuildTask)
+  - `Service` (generic class type that has no parent class)
+Some parent classes have unorthodox naming, in which case suffix with something more standard
+  - `Page` (SiteTree)
+  - `Controller` (LeftAndMain, CMSMain)
+- Do not suffix classes if they are a subclass any of the following:
+  - `DataObject`
+- Suffix interfaces with `Interface`
+- Suffix traits with `Trait`
+- Prefix abstract classes with `Abstract`
+
+### Other naming conventions
+
+- Extensions should match the name of the class they are extending. For example, an extension class that's added to LeftAndMain should be called LeftAndMainExtension
+- When choosing names for variables and methods use the guiding principle "make it as obvious as possible"
+- Ensure that variable names and method naming is consistent. For example, if you are creating a new class and have a variable named `$itemValue`` which is passed to another method in the same class, the paramater name for that method should also be `$itemValue``.
+
+## Directory naming - src dir
+
+- Put similar types of classes in the same directory. For example, put Controllers in a Controllers directory
+- Put DataObjects in a directory called Models
+- Do not group different class types together in a directory relating to a single feature
+- Use initial caps pluralised for the directory naming e.g. use `Controllers` rather than `Controller` or `controllers`
```

Comment:
Seeing it non-pluralised hurts my brain

If it's a directory of many controllers, it should call it "Controllers". Calling it the singular Controller is just grammatically incorrect

Reply:
If that's the only reason, I'd prefer singular as it reads nicer in the namespace (`SilverStripe\Controller\SomeSpecialController`)

Symfony also tends to use singular, so if we're aiming to start using more and more symfony code, it might be nice to align with their namespace naming convention in this case.

That said, if nobody else chimes in, then I'm not too fussed.

Reply:
Yeah probably reads a little better in the namespace.  Though I don't tend to look at the namespace much though I see directories all the time

Symfony also has yoda conditionals, so I'm too not sure if we should blindly follow symfony on things :-)

Reply:
I'm not saying "blindly" follow. In this case, we would have our use statements next to use statements for symfony stuff, so it makes sense in this case that they would be similar.

```php
use Symfony/Controller/SomeController;
use SilverStripe/Controller/SomeOtherController;
```

But yeah, directories are _or should be_ namespaces - we want to move more towards a purer PSR12, so we should always keep namespaces in mind when dealing with directories.

Reply:
Had an offline chat, agreed on plural